src/components/taskDetails/index.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { ChangeEvent, FC } from 'react';
+import NavBar from '@/components/navBar/index';
+import TaskContainer from './TaskContainer';
+import Details from './Details';
+import classNames from './task-details.module.scss';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import task from '@/interfaces/task.type';
+import { useState, useEffect, useContext } from 'react';
+import useFetch from '@/hooks/useFetch';
+import { isUserAuthorizedContext } from '@/context/isUserAuthorized';
+import fetch from '@/helperFunctions/fetch';
+
+type Props = {
+  url: string;
+  taskID: string;
+};
+
+const TaskDetails: FC<Props> = ({ url, taskID }) => {
+  const isAuthorized = useContext(isUserAuthorizedContext);
+  const [isEditing, setIsEditing] = useState(false);
+  const [initialData, setInitialData] = useState<task>();
+  const [taskDetails, setTaskDetails] = useState<task>();
+  const [editedDetails, setEditedDetials] = useState({});
+  const { response, error, isLoading } = useFetch(url);
+  const { SUCCESS, ERROR } = ToastTypes;
+
+  useEffect(() => {
+    const fetchedData: task = { ...response.taskData };
+    setTaskDetails(fetchedData);
+    setInitialData(fetchedData);
+  }, [isLoading, response]);
+
+  function convertTimeStamp(timeStamp: string) {
+    const dateTime = new Date(parseInt(timeStamp));
+    return dateTime.toLocaleString();
+  }
+
+  function renderTextarea(name: string, value: string) {
+    return (
+      <textarea
+        className={classNames['textarea']}
+        name={name}
+        value={value}
+        onChange={(event) => handleChange(event)}
+      />
+    );
+  }
+
+  function handleChange(event: ChangeEvent) {
+    const formData: any = {
+      [(event.target as HTMLInputElement).name]: (
+        event.target as HTMLInputElement
+      ).value,
+    };
+    setEditedDetials({ ...editedDetails, ...formData });
+    setTaskDetails({ ...taskDetails, ...formData });
+  }
+
+  function onCancel() {
+    setIsEditing(false);
+    setEditedDetials({});
+    setTaskDetails(initialData);
```

Comment:
isn't this changed to task details data fetched, in line 30?

Reply:
That is for loading the data that is fetched from the API. In this case when superuser clicks on cancel, we are resetting the data so that the UI can be restored